File: task3.cs
using System;

class Program
{
    static int Diff(int a, int b) => a > b ? a - b : b - a;

    static int Prod(int n)
    {
        int res = 1;
        while (n > 0)
        {
            res *= n % 10;
            n /= 10;
        }
        return res;
    }

    static int SumDigits(int n)
    {
        int res = 0;
        while (n > 0)
        {
            res += n % 10;
            n /= 10;
        }
        return res;
    }

    static void Main()
    {
        Console.Write("Enter the number: ");
        if (int.TryParse(Console.ReadLine(), out int num) && num > 0)
        {
            int summary = SumDigits(num);
            int product = Prod(num);
            Console.WriteLine($"{summary} {product} {Diff(summary, product)}");
        }
        else
        {
            Console.WriteLine("Number is not natural");
        }
    }
}

File: task3.kt
fun diff(a: Int, b: Int) = if (a > b) a - b else b - a

fun prod(n: Int): Int {
    var num = n
    var res = 1
    while (num > 0) {
        res *= num % 10
        num /= 10
    }
    return res
}

fun sumDigits(n: Int): Int {
    var num = n
    var res = 0
    while (num > 0) {
        res += num % 10
        num /= 10
    }
    return res
}

fun main() {
    print("Enter the number: ")
    val num = readLine()?.toIntOrNull() ?: 0
    if (num > 0) {
        val summary = sumDigits(num)
        val product = prod(num)
        println("$summary $product ${diff(summary, product)}")
    } else {
        println("Number is not natural")
    }
}